File: Coding/DDA.py
import matplotlib.pyplot as plt
import math


def lineDDA(xa, ya, xb, yb):
    dx, dy, x, y = xb - xa, yb - ya, xa, ya
    if math.fabs(dx) > math.fabs(dy):
        steps = math.fabs(dx)
    else:
        steps = math.fabs(dy)
    xinc, yinc, lx, ly = (dx / steps), (dy / steps), [round(x), ], [round(y), ]
    for k in range(0, int(steps)):
        x, y = x + xinc, y + yinc
        lx.append(round(x))
        ly.append(round(y))
    ax, ay = (sum(lx) / len(lx)), (sum(ly) / len(ly))
    lax, lay = [lx[0], round(ax), lx[-1]], [ly[0], round(ay), ly[-1]]
    plt.scatter(lx, ly)
    plt.plot(lax, lay, color='green')
    plt.show()


xa, ya, xb, yb = int(input("Enter 1st x-coordinate : ")), int(input("Enter 1st y-coordinate : ")), int(input("Enter 2nd x-coordinate : ")), int(input("Enter 2nd y-coordinate : "))
lineDDA(xa, ya, xb, yb)

File: Coding/DDA.c
#include<math.h>
#include<stdio.h>
#include<conio.h>
#include<graphics.h>

#define ROUND(a) ((int)(a+0.5))

void lineDDA(int xa, int xb, int ya, int yb)
{
	int dx,dy,steps,k;
	float xinc,yinc,x,y;
	dx=xb-xa;
	dy=yb-ya;
	x=xa;
	y=ya;
	
	if (abs(dx)>abs(dy))
		steps=abs(dx);
	else
		steps=abs(dy);
	xinc=dx/(float)steps;
	yinc=dy/(float)steps;
	
	setPixel(ROUND(x),ROUND(y));
	for(k=0;k<steps;k++)
	{
		x+=xinc;
		y+=yinc;
		setPixel(ROUND(x),ROUND(y));
	}
}
void main()
{
	int xa,xb,ya,yb;
	printf("Enter 1st endpoint :");
	scanf("%d%d",&xa,&ya);
	printf("\nEnter 2nd endpoint :");
	scanf("%d%d",&xb,&yb);
	printf("\nThe line :");
	lineDDA(xa,xb,ya,yb);
	getch();
}
	
	
